x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_list.tsx

```diff
@@ -144,7 +147,9 @@ export const DataStreamList: React.FunctionComponent<RouteComponentProps<MatchPa
 
         <DataStreamTable
           filters={
-            dataStreamName !== undefined ? `name=${decodePathFromReactRouter(dataStreamName)}` : ''
+            isDeepLink && dataStreamName !== undefined
```

Comment:
Suggestion: I see the `dataStreamName !== undefined` logic existed before, but it might be worth changing here. This check is actually a lot more permissive than it needs to be (i.e., it tests true for all non-`undefined` values for dataStreamName). I would suggest `dataStreamName != undefined`.

No need to change if we have 100% confidence that `dataStreamName` will always be either a string or undefined.

Reply:
Good point, JL! Yes, I think we can be 100% confident it will always be either a string or undefined. It's provided by react router and is extracted from the path, so I'm not seeing any opportunities for it to be null, a number, or an object. I think we can leave as-is.